ruby/ql/src/utils/modeleditor/FrameworkModeAccessPaths.ql

```diff
@@ -0,0 +1,119 @@
+    result = paramNode.getName()
+  )
+  or
+  exists(DataFlow::MethodNode methodNode, string name |
+    methodNode.getKeywordParameter(name) = paramNode
+  |
+    result = name + ":"
+  )
+  or
+  exists(DataFlow::MethodNode methodNode | methodNode.getBlockParameter() = paramNode |
+    result = "&" + paramNode.getName()
+    or
+    not exists(paramNode.getName()) and result = "&"
+  )
+  or
+  exists(DataFlow::MethodNode methodNode | methodNode.getSelfParameter() = paramNode |
+    result = "self"
+  )
+  or
+  exists(DataFlow::MethodNode methodNode | methodNode.getHashSplatParameter() = paramNode |
+    result = "**" + paramNode.getName()
+    or
+    not exists(paramNode.getName()) and result = "**"
+  )
+}
+
+predicate simpleParameters(string type, string path, string value, string details) {
+  exists(DataFlow::MethodNode methodNode, DataFlow::ParameterNode paramNode |
+    methodNode.getLocation().getFile() instanceof Util::RelevantFile and
+    (
+      // Check that this parameter belongs to this method
+      // Block parameter explicitly excluded because it's already included
+      // as part of the blockArguments predicate
+      paramNode = Util::getAnyParameter(methodNode) and
+      paramNode != methodNode.getBlockParameter()
+    )
+  |
+    Util::pathToMethod(methodNode, type, path) and
+    value = Util::getArgumentPath(paramNode) and
+    details = parameterDetails(paramNode)
+  )
+}
+
+predicate blockArguments(string type, string path, string value, string details) {
+  exists(DataFlow::MethodNode methodNode, DataFlow::CallNode callNode |
+    methodNode.getLocation().getFile() instanceof Util::RelevantFile and
+    callNode = methodNode.getABlockCall()
+  |
+    (
+      exists(DataFlow::VariableAccessNode argNode, int i |
+        argNode = callNode.getPositionalArgument(i)
+      |
+        value = "Argument[block].Parameter[" + i + "]" and
+        details = argNode.asVariableAccessAstNode().getVariable().getName()
+      )
+      or
+      exists(DataFlow::ExprNode argNode, string keyword |
+        argNode = callNode.getKeywordArgument(keyword)
+      |
+        value = "Argument[block].Parameter[" + keyword + ":]" and
+        details = ":" + keyword
+      )
+      or
+      value = "Argument[block]" and
+      (
+        details = callNode.getMethodName()
+        or
+        not exists(callNode.getMethodName()) and
+        callNode.getExprNode().getExpr() instanceof YieldCall and
+        details = "yield ..."
+      )
+    ) and
+    Util::pathToMethod(methodNode, type, path)
+  )
+}
+
+predicate returnValue(string type, string path, string value, string details) {
+  exists(DataFlow::MethodNode methodNode, DataFlow::Node returnNode |
+    methodNode.getLocation().getFile() instanceof Util::RelevantFile and
+    returnNode = methodNode.getAReturnNode()
+  |
+    Util::pathToMethod(methodNode, type, path) and
+    value = "ReturnValue" and
+    details = returnNode.toString()
```

Comment:
I'm not sure how to remove this warning without re-implementing a large part of the CodeQL libraries. Explicit returns are easy to support by just using `return` (since this is what the libraries also do). Implicit returns are much harder to support, like showing `yield ...` in the tests.

Reply:
For my own understanding: what is `details` used for in the editor? If we were to omit this column then I think we could avoid this warning.

Reply:
Nice. OK, well I think the QL4QL warning is informational rather than blocking, so let's proceed with the query as-is.

Reply:
Actually I've just had an idea about this - if you change the query so that it returns an `Expr` for `details` (and maybe rename it) then we'll still get the string value but that'll also make QL4QL happy (and we get a bonus source location as well, which can be ignored if not needed).

Reply:
I've now changed it to return a `DataFlow::Node` and renamed the column to `node` to reflect this. There are no changes to the string representation, so we'll definitely still be able to use this in VS Code.